src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java

```diff
@@ -114,10 +147,75 @@ public void finishProcessing() {
         super.finishProcessing();
         final Map<String, Set<File>> propFilesMap =
             arrangePropertyFiles(propertyFiles, basenameSeparator);
+        if (!checkedLanguageCodes.isEmpty()) {
+            checkExistenceOfTranslationFiles(propFilesMap);
+        }
         checkPropertyFileSets(propFilesMap);
     }
 
     /**
+     * Checks existence of the translation files (arranged in a map)
+     * for each resource bundle in project.
+     * @param translationFilesMap the translation files bundles organized as Map.
+     */
+    private void checkExistenceOfTranslationFiles(Map<String, Set<File>> translationFilesMap) {
+        final Set<Entry<String, Set<File>>> resourceBundles = translationFilesMap.entrySet();
+        for (Entry<String, Set<File>> bundle : resourceBundles) {
+            final SortedSet<File> filesInResourceBundle = new TreeSet<>(bundle.getValue());
```

Comment:
Can you explain why we need to create one more set here? Can't it be proper `TreeSet` from the very beginning?